### PR TITLE
Vcluster/2965 claim automation auth

### PR DESCRIPTION
**Pull request by TopSwagCode**

### CODE REVIEW

File: domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomationControllerTest.cs [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
-namespace Worker.IntegrationTests.Api.Controllers;
+namespace API.IntegrationTests.ClaimAutomation.Api.Controllers;
 
-public class ClaimAutomationControllerTest(ClaimAutomationApplicationFactory factory)
-    : IClassFixture<ClaimAutomationApplicationFactory>
+[Collection(IntegrationTestCollection.CollectionName)]
+public class ClaimAutomationControllerTest
 {
+    private readonly string sub = Guid.NewGuid().ToString();
+    //private readonly string tin = "12345678";
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomationControllerTest.cs:14]: delete?

File: domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomation20240515ControllerTest.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+using System.Net;
+using System.Threading.Tasks;
+using API.IntegrationTests.Factories;
+using FluentAssertions;
+using Xunit;
+
+namespace API.IntegrationTests.ClaimAutomation.Api.Controllers;
+
+[Collection(IntegrationTestCollection.CollectionName)]
+public class ClaimAutomation2030ControllerTest
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomation20240515ControllerTest.cs:11]: I generally prefer a test class to be named after the class it tests, so ex. ClaimAutomation20240515Controller**Test**. I think the file is named like this?

File: domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomation20240515Controller.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        try
+        {
+            claim = await unitOfWork.ClaimAutomationRepository.AddClaimAutomationArgument(claimAutomationArgument);
+            var claimAutomationArgumentDto = new ClaimAutomationArgumentDto(claim.CreatedAt.ToUnixTimeSeconds());
+
+            return CreatedAtAction(nameof(GetClaimAutomation), null, claimAutomationArgumentDto);
+        }
+        catch (DbUpdateException)
+        {
+            var claimAutomation = await unitOfWork.ClaimAutomationRepository.GetClaimAutomationArgument(organizationId);
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomation20240515Controller.cs:54]: This is probably just copied from the old implementation, but what is this handling?
Concurrent updates to the same ClaimAutomationArgument table for same organizationId? In this case it just assumes that an error updating the table must mean the row has been created in the mean time, in my opinion we should return an error, and allow the user to retry the action.

File: domains/transfer/ClaimAutomation/Worker/Program.cs [hunk 4]
```diff
@@ -42,16 +40,11 @@
         providerOptions => providerOptions.EnableRetryOnFailure()
     ),
     optionsLifetime: ServiceLifetime.Singleton);
-builder.Services.AddDbContextFactory<ApplicationDbContext>();
+builder.Services.AddDbContextFactory<ApplicationDbContext>(); // TODO: Look into why we add DbContext twice.
```
Comment (sahma19) [inline domains/transfer/ClaimAutomation/Worker/Program.cs:43]: Fix this, or remove the TODO

**Review by sahma19 — APPROVED** [review]

File: domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs [hunk 5]
```diff
@@ -42,6 +45,11 @@ public ITransferAgreementProposalRepository TransferAgreementProposalRepo
         get => transferAgreementProposalRepo ??= new TransferAgreementProposalRepository(context);
     }
 
+    public IClaimAutomationRepository ClaimAutomationRepository
+    {
+        get => claimAutomationRepository ??= new ClaimAutomationRepository(context);
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs:50]: ??= FTW :)

**Review by martinhenningjensen — CHANGES_REQUESTED** [review]

File: domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomation20240515Controller.cs (hunk 3 above)
Comment (TopSwagCode) [reply domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomation20240515Controller.cs:54]: This is just same logic as is. Don't know why it's done :D

File: domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs (hunk 5 above)
Comment (TopSwagCode) [reply domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs:50]: Copy paste of logic as the other places. To stay consistent

File: domains/transfer/ClaimAutomation/Worker/Program.cs (hunk 4 above)
Comment (TopSwagCode) [reply domains/transfer/ClaimAutomation/Worker/Program.cs:43]: removed

File: domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomation20240515ControllerTest.cs (hunk 2 above)
Comment (TopSwagCode) [reply domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomation20240515ControllerTest.cs:11]: ahh yeah. Forgot to update this after I trolled you with versions :D will fix.

File: domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomationControllerTest.cs (hunk 1 above)
Comment (TopSwagCode) [reply domains/transfer/Transfer.API/API.IntegrationTests/ClaimAutomation/Api/Controllers/ClaimAutomationControllerTest.cs]: it is done :D

**Review by martinhenningjensen — APPROVED** [review]

**Pull request merged**